components/director/internal/repo/common.go

```diff
@@ -11,32 +11,44 @@ type Condition struct {
 	Val   string
 }
 
-func getAllArgs(tenant string, conditions Conditions) []interface{} {
-	allArgs := []interface{}{tenant}
+func getAllArgs(tenant *string, conditions Conditions) []interface{} {
+	allArgs := []interface{}{}
+	if tenant != nil {
+		allArgs = append(allArgs, tenant)
+	}
 	for _, idAndVal := range conditions {
 		allArgs = append(allArgs, idAndVal.Val)
 	}
 	return allArgs
 }
 
-func appendEnumeratedConditions(query string, startIdx int, conditions Conditions) string {
-	out := query
+func appendEnumeratedConditions(builder *strings.Builder, startIdx int, conditions Conditions) *strings.Builder {
```

Comment:
It is a bit confusing. When you return a builder then why you modify it? In this implementation, I would expect that you do not modify the builder which is passed as a parameter - see the append implementation from Go.